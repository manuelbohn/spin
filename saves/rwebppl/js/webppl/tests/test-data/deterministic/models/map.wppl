var inc = function(x) {
  return x + 1
};

var paste = function(x, y) {
  return x + '-' + y;
};

var plus = function(x, y) {
  return x + y;
};

var eq = function(x, y) {
  return x === y;
};

[map(inc, []),
 map(inc, [0]),
 map(inc, [0, 1]),
 map(inc, [0, 1, 2]),
 map(inc, [0, 1, 2, 3]),
 map(inc, [0, 1, 2, 3, 4]),
 map2(eq, [], []),
 map2(eq, [0, 1, 2], [0, 1, 2]),
 map2(plus, [0, 1, 2], [0, 1, 2]),
 map2(plus, repeat(50000, function() { return 1}), repeat(50000, function() { return 2}))[49999],
 sum(map(inc, repeat(50000, function(x) { return 0 }))),
 sum(mapN(function(i) { return i }, 50000)),
 mapIndexed(inc, []),
 mapIndexed(paste, ['a', 'b', 'c', 'd']),
 mapIndexed(paste, mapN(function(i) { return i},50000))[49999],
 map(function(ar) { return map(inc, ar) }, [[0, 1, 2, 3],
   [4, 5, 6, 7],
                                            [8, 9, 10, 11]]),
 mapData({data: []}, inc),
 mapData({data: [0, 1, 2]}, inc)
]
