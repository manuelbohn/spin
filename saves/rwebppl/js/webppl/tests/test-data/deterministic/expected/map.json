{
  "result": [
    [],
    [1],
    [1,2],
    [1,2,3],
    [1,2,3,4],
    [1,2,3,4,5],
    [],
    [true,true,true],
    [0,2,4],
    3,
    50000,
    1249975000,
    [],
    ["0-a","1-b","2-c","3-d"],
    "49999-49999",
    [[1,2,3,4],
     [5,6,7,8],
     [9,10,11,12]],
    [],
    [1,2,3]
  ]
}
